File: src/ServiceControl.Monitoring/Http/Diagrams/MonitoredEndpoint.cs
namespace ServiceControl.Monitoring.Http.Diagrams
{
    using System.Collections.Generic;

    public class MonitoredEndpoint
    {
        public string Name { get; set; }
        public bool IsStale { get; set; }
        public string[] EndpointInstanceIds { get; set; } = [];
        public Dictionary<string, MonitoredValues> Metrics { get; set; } = []; // HINT: System.Text.Json requires public setters een to serialize properties
        public int DisconnectedCount { get; set; }
        public int ConnectedCount { get; set; }
    }
}

File: src/resources/MonitoredEndpoint.ts
export default interface MonitoredEndpoint {
  Name: string;
  IsStale: boolean;
  EndpointInstanceIds: string[];
  Metrics: { [key: string]: MonitoredValues };
  DisconnectedCount: number;
  ConnectedCount: number;
}

export interface MonitoredValues {
  Average?: number;
  Points: number[];
}
